File: hw/viteri/u2/HW18ZooInPython/Zoo/model/Mammal.py
from datetime import date
from model.Animal import Animal
from model.Cage import Cage

class Mammal(Animal):
    def __init__(self, id, description, born_on_date, gender, cage, number_of_bones, vertebrate):
        super().__init__(id, description, born_on_date, gender, cage, number_of_bones, vertebrate)
    
    def feed(self, food_amount):
        print("Feeding the mammal with", food_amount, "units of food")

File: hw/segarra/u2/HW18ZooInPythonAndJavaScript/ec.edu.zooinjavascript.model/Mammal.js
import Animal from "./Animal.js";

export default class Mammal extends Animal{
    constructor(id, description, bornOnDate, gender, cage, numberOfBones, vertebrate){
        if (new.target === Mammal) {
            throw new TypeError("Cannot construct Mammal instances directly");
        }
        super(id, description, bornOnDate, gender, cage, numberOfBones, vertebrate)
    }
}